elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -140,11 +140,34 @@ public void readAndValidateConfigs() throws IOException {
         validateRequiredConfigsProvided();
         validateNameUniqueness(this.elideSQLDBConfig.getDbconfigs());
         validateNameUniqueness(this.elideTableConfig.getTables());
+        validateInheritanceCycles(this.elideTableConfig);
         populateInheritance(this.elideTableConfig);
         validateTableConfig(this.elideTableConfig);
         validateJoinedTablesDBConnectionName(this.elideTableConfig);
     }
 
+    private void validateInheritanceCycles(ElideTableConfig tables) {
+        tables.getTables().stream().forEach(table -> {
+            validateInheritanceCycles(tables, table, new HashSet<>());
+        });
+    }
+
+    private void validateInheritanceCycles(ElideTableConfig tables, Table table, Set<Table> visited) {
+        visited.add(table);
+
+        if (!table.hasParent()) {
+            return;
+        }
+        Table parent = table.getParent(tables);
```

Comment:
table.hasParent() doesn't verify if parent model is actually defined so parent can become `null` here.

Reply:
may be move this logic to `validateExtend` method.